File: backend/app/models/task.py
from sqlalchemy import Column, Integer, String, Text, DateTime, JSON
from sqlalchemy.sql import func
from app.core.database import Base

class Task(Base):
    __tablename__ = "tasks"
    
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255), nullable=False)
    description = Column(Text, nullable=False)
    expected_output = Column(Text, nullable=False)
    tools = Column(JSON, default=[])
    context = Column(JSON, default={})
    additional_params = Column(JSON, default={})
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now()) 

File: frontend/types/task.ts
export interface Task {
  id: number;
  name: string;
  description: string;
  expected_output: string;
  tools: any[];
  context: Record<string, any>;
  additional_params: Record<string, any>;
  created_at: string;
  updated_at: string;
}

export interface TaskCreate {
  name: string;
  description: string;
  expected_output: string;
  tools?: any[];
  context?: Record<string, any>;
  additional_params?: Record<string, any>;
}

export interface TaskUpdate {
  name?: string;
  description?: string;
  expected_output?: string;
  tools?: any[];
  context?: Record<string, any>;
  additional_params?: Record<string, any>;
}

export interface TaskFilters {
  search?: string;
  skip?: number;
  limit?: number;
} 